oak/server/rust/oak_runtime/src/node/roughtime.rs

```diff
@@ -16,32 +16,170 @@
 
 //! Roughtime client pseudo-Node functionality.
 
-use crate::runtime::RuntimeProxy;
-use log::{info, warn};
+use crate::{
+    io::Receiver,
+    node::grpc::invocation::Invocation,
+    runtime::RuntimeProxy,
+    time::{
+        get_default_servers, RoughtimeClient, RoughtimeServer, DEFAULT_MAX_RADIUS_MICROSECONDS,
+        DEFAULT_MIN_OVERLAPPING_INTERVALS, DEFAULT_SERVER_RETRIES, DEFAULT_TIMEOUT_SECONDS,
+    },
+};
+use log::{debug, error, info};
+use oak_abi::{
+    proto::{
+        google::rpc::Code,
+        oak::{
+            application::RoughtimeClientConfiguration,
+            encap::GrpcResponse,
+            roughtime::{RoughtimeRequest, RoughtimeResponse},
+        },
+    },
+    OakStatus,
+};
+use prost::Message;
 use tokio::sync::oneshot;
 
 /// Roughtime client pseudo-Node.
 pub struct RoughtimeClientNode {
     node_name: String,
+    client: RoughtimeClient,
 }
 
 impl RoughtimeClientNode {
     /// Creates a new [`RoughtimeClientNode`] instance, but does not start it.
-    pub fn new(node_name: &str) -> Self {
+    pub fn new(node_name: &str, config: &RoughtimeClientConfiguration) -> Self {
+        let timeout_seconds = if config.timeout_seconds == 0 {
+            DEFAULT_TIMEOUT_SECONDS
+        } else {
+            config.timeout_seconds as u64
+        };
```

Comment:
If these values are meant to be optional, then perhaps should use proper types in the protos, e.g. https://developers.google.com/protocol-buffers/docs/reference/csharp/class/google/protobuf/well-known-types/u-int32-value, which prost should convert into `Option<u32>` , which you can then use as `config.server_retries.unwrap_or(DEFAULT_SERVER_RETRIES)` etc.

Reply:
Done.